### ISSUE TITLE
Vercel SSR error with dynamic routes

### ISSUE THREAD

**Issue by nathanlawsn** [withastro/astro#8407]
### Astro Info

```block
Astro                    v3.0.8
Node                     v16.20.0
System                   Linux (x64)
Package Manager          unknown
Output                   hybrid
Adapter                  @astrojs/vercel/serverless
Integrations             none
```

### Describe the Bug

Dynamic routes throw an error when deploying to Vercel.

https://astro-vercel-ssr-error.vercel.app/
https://astro-vercel-ssr-error.vercel.app/blog/blog-post-1

```
[ssr] Error: Expected an exported Astro component but received typeof undefined
    at renderPage (file:///var/task/vercel/path0/dist/chunks/entrypoint_eea6d45b.mjs:579:11)
    at #tryRenderRoute (file:///var/task/vercel/path0/dist/chunks/entrypoint_eea6d45b.mjs:774:24)
    at SSRRoutePipeline.renderRoute (file:///var/task/vercel/path0/dist/chunks/entrypoint_eea6d45b.mjs:721:46)
    at App.render (file:///var/task/vercel/path0/dist/chunks/entrypoint_eea6d45b.mjs:934:39)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.handler (file:///var/task/vercel/path0/dist/chunks/entrypoint_eea6d45b.mjs:1317:33)
    at async Server.<anonymous> (/var/task/___vc/__launcher.js:82:14)
```

It appears if you have multiple files with the same filename (e.g. [slug].astro) yet in separate directories then the dynamic route fails.

### What's the expected result?

I should be able to have multiple dynamic routes under different directories. e.g. /blog

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-yrtccw

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by ematipico** [comment]
Is this a duplicate of https://github.com/withastro/astro/issues/8401?

**Comment by lilnasy** [comment]
Confirmed fixed by #8408
